docs/src/main/asciidoc/elytron-properties-guide.adoc

```diff
@@ -0,0 +1,168 @@
+    </dependency>
+</dependencies>
+----
+
+== Configuration
+
+The elytron-security-properties-file extension currently supports two different realms for the storage of authentication
+and authorization information. Both support storage of this information in properties type files. The next two sections detail the specific configuration properties.
+
+=== Property Files Realm Configuration
+
+The property files realm supports mapping of users to password and users to roles with a combination of properties files. To enable and configure it, the following configuration properties are used:
+
+|===
+|Property Name|Default|Description
+|quarkus.users.file.enabled|false|Determine whether security via the file realm is enabled
+|quarkus.users.file.auth-mechanism|BASIC|Name of authentication mechanism to use
+|quarkus.users.file.realm-name|Quarkus|Name to assign the security realm
+|quarkus.users.file.users|users.properties|Classpath resource name of properties file containing user to password mappings; see <<Users.properties>>
+|quarkus.users.file.roles|roles.properties|Classpath resource name of properties file containing user to role mappings; see <<Roles.properties>>
+|===
+
+.example application.properties file section for property files realm
+[source,properties]
+----
+quarkus.users.file.enabled=true
+quarkus.users.file.users=test-users.properties
+quarkus.users.file.roles=test-roles.properties
+quarkus.users.file.auth-mechanism=BASIC
+quarkus.users.file.realm-name=MyRealm
+quarkus.users.file.plain-test=true
+----
+
+==== Users.properties
+
+The `quarkus.users.file.users` configuration property specifies a classpath resource which is a properties file with a user to password mapping, one per line. The following <<test-users-example>> illustrates the format:
+
+[#test-users-example]
+.example test-users.properties file
+[source,properties]
+----
+scott=jb0ss <1>
+jdoe=p4ssw0rd <2>
+stuart=test
+noadmin=n0Adm1n
+----
+<1> User `scott` has password defined as `jb0ss`
+<2> User `jdoe` has password defined as `p4ssw0rd`
+
+This file has the usernames and passwords stored in plain text, which is not recommended. If plain-text is set to false
+(or omitted) in the config then passwords must be stored in the form `MD5 ( username : realm : password )`. This can
+be generated for the first example above by running the command `echo -n scott:MyRealm:jb0ss | md5` from the command line.
+
+==== Roles.properties
+
+.example test-roles.properties file
+[source,properties]
+----
+scott=Admin,admin,Tester,user <1>
+jdoe=NoRolesUser <2>
+stuart=admin,user <3>
+noadmin=user
+----
+<1> User `scott` has been assigned the roles `Admin`, `admin`, `Tester` and `user`
+<2> User `jdoe` has been assigned the role `NoRolesUser`
+<3> User `stuart` has been assigned the roles `admin` and `user`.
+
+Given these role mappings, only user `scott` would be allowed to access the `/subject/secured` endpoint from the <<subject-example>>.
```

Comment:
The `<<subject-example>>` cross reference points to nothing existing in this document.